### ISSUE TITLE
caret with Generalized Random Forest grf ?

### ISSUE THREAD

**Issue by engie-elkhait-meine** [topepo/caret#878]
Hello !

I am working with the grf package which is quite interesting 👍 Could you in the next releases integrate the method (regression_forest) in caret ?! 

Thanks

**Comment by topepo** [comment]
Perhaps. If you are really interested in it, you could submit a pull request and base your code on the existing model code.

**Comment by hadjipantelis** [comment]
@engie-elkhait-meine Consider using Python directly. The R package itself is a wrapper around the Python package which itself is a wrapper of the C++ code. Unless you are doing this for fun (eg. Kaggle - no money, job-security involved), they will be so many moving components that it is not worth to view this as something other than a proof-of-concept implementation. Sadly. 

(I say this as someone who wrote a quick implementation for `RGF` in `caret` in the past but  was so annoyed by the total dependency on Python that decided against pushing it. The proper solution would be writing a package that uses Rcpp to bind to the RGF C++ code directly taking Python out of the equation.)

**Comment by MarkusBerroth** [comment]
I thought the same and started implementing it in caret.
But I am not sure if it is useful at all.
Because one of the main advantages of caret is the tuning of the model parameters with grid or random search, but grf has its own tuning function with "tune_regression_forest".
Also for "causal_forest" it doesn't really makes sense, because caret only considers a Y and X input and no third input for W, respectively caret is designed for making oos predictions and not for estimating heterogenous treatment effects.

**Comment by engie-elkhait-meine** [comment]
Thanks a lot ! 

The main reason why I wanted to have the regression_forest method in the caret package is for prediction intervals which is not too long like for example other methods estimating it (browsing each tree)

**Comment by MarkusBerroth** [comment]
Hey, 
I implemented the grf in caret for my own use.
https://github.com/MarkusBerroth/caret-grf/blob/master/grf.R
It takes the output of `tune_regression_forest()` for the grid-search. 
Also honest tree splitting is turned on as default, because it is needed for consistent variance estimation. 
On the other side I tried my implementation on 4 different common available datasets and the RMSE was lower with honest = FALSE. Also the results were worse than the random forest implementations of the randomforest and ranger package. For this "test" I used  a random-search for the tuning parameters around the results of `tune_regression_forest()`
So probably there is room for improvement of the default values.
